smack-experimental/src/main/java/org/jivesoftware/smackx/reactions/ReactionsListener.java

```diff
@@ -0,0 +1,32 @@
+/**
+ *
+ * Copyright 2025 Ismael Nunes Campos
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.reactions;
+
+import org.jivesoftware.smack.packet.Message;
+
+import org.jivesoftware.smackx.reactions.element.ReactionsElement;
+
+public interface ReactionsListener {
+
+    /**
+     * Listener method that gets called when a {@link Message} containing a {@link ReactionsElement} is received.
+     * @param reactionsElement reactionsElement
+     * @param message message
+     */
+    void onReactionReceived(Message message, ReactionsElement reactionsElement);
```

Comment:
Let's make the API of `ReactionsListener` more user friendly. The user is primary interested in two aspects: the ID of the message that was reacted upon, and the reactions itself. Everything else is secondary.
```suggestion
    void onReactionReceived(String id, List<String> reactions, ReactionsElement reactionsElement, Message message);
```

Also make sure to document what `message` is. Is it the Message the reaction arrived with, or is the the message on which was reacted on?

Reply:
Why was this marked as resolved if it wasn't resolved?